### PR TITLE
[GlobalISel] Handle more commutable in `commute_constant_to_rhs`

### PR DESCRIPTION
**Pull request by darkbuck**
- Fix the matcher, which results in an infinite loop if both LHS and RHS
  are constants.
- Update test cases.

### CODE REVIEW

File: llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp [hunk 1]
```diff
@@ -6276,14 +6276,15 @@ bool CombinerHelper::matchShiftsTooBig(MachineInstr &MI) {
...
-    return false;
+  if (!getIConstantVRegVal(LHS, MRI)) {
+    // Skip if LHS is not a constant. But, LHS may be a
+    // G_CONSTANT_FOLD_BARRIER. If so we commute as long as we don't already
+    // have a constant on the RHS.
+    if (MRI.getVRegDef(LHS)->getOpcode() !=
+        TargetOpcode::G_CONSTANT_FOLD_BARRIER)
+      return false;
+  }
+  // Commute as long as RHS is not a constant or G_CONSTANT_FOLD_BARRIER.
```
Comment (arsenm) [inline llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp:6287]: Should do this part in a separate PR from the add to the list of opcodes

**Comment by darkbuck** [comment]
That infinite loop issue is found in `test/CodeGen/AArch64/GlobalISel/prelegalizer-combiner-divrem-insertpt-conflict.mir`, which has `G_UMULH` with 2 constant operands.

File: llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp (hunk 1 above)
Comment (darkbuck) [reply llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp:6287]: Check PR#87426 and PR#87424
